plugins/module_utils/nd.py

```diff
@@ -301,6 +301,10 @@ def request(self, path, method=None, data=None, file=None, qs=None, prefix="", f
                     self.fail_json(msg="ND Error {code}: {message}".format(**payload), data=data, info=info, payload=payload)
                 elif "messages" in payload and len(payload.get("messages")) > 0:
                     self.fail_json(msg="ND Error {code} ({severity}): {message}".format(**payload["messages"][0]), data=data, info=info, payload=payload)
+                elif "errors" in payload:
+                    if ignore_not_found_error:
+                        return {}
```

Comment:
The reason, this was repeated here is because -> 
- in nd_federation member module
  - federation has to be created before creating the federation member 
  - the first call this module makes is a query to verify if the federation was added and it will return "errors" in payload, which means it will exit from the module but we want it to continue execution since we can add federation later 
  - if ignore_not_found_error: -> works as an additional check and lets us continue the execution of the code